metis_client/models/calculation.py

```diff
@@ -0,0 +1,43 @@
+"""Calculation models"""
+
+from dataclasses import dataclass
+from typing import Optional, Sequence
+
+from .datasource import MetisDataSourceDTO, MetisDataSourceModel
+from .timestamp import MetisTimestampsDTO, MetisTimestampsModel
+
+
+class MetisCalculationDTO(MetisTimestampsDTO):
+    "Calculation DTO"
+    id: int
+    name: str
+    userId: int
+    status: int
+    progress: int
+    result: Optional[Sequence[MetisDataSourceDTO]]
+
+
+@dataclass(frozen=True)
+class MetisCalculationModel(MetisTimestampsModel):
+    "Calculation model"
+    id: int
+    name: str
+    user_id: int
+    status: int
+    progress: int
+    result: Sequence[MetisDataSourceModel]
+
+    @classmethod
+    def from_dto(cls, dto: MetisCalculationDTO) -> "MetisCalculationModel":
```

Comment:
The similar`from_dto` method is used across many other places; would it make sense to add an inheritable mixture maybe instead?

Reply:
But all `from_dto`'s are different.